packages/admin-stories/src/admin/table/TableWithFilterBar.tsx

```diff
@@ -0,0 +1,213 @@
+    users: Array<{
+        id: number;
+        name: string;
+        username: string;
+        email: string;
+        phone: string;
+        website: string;
+    }>;
+}
+
+interface IFilterValues {
+    username: string;
+    name: string;
+    email: string;
+    website: string;
+    range: {
+        minMaxArea: {
+            min: number;
+            max: number;
+        };
+        minMaxHeight: {
+            min: number;
+            max: number;
+        };
+    };
+}
+
+interface IVariables extends IFilterValues {
+    pathFunction: any;
+}
+
+const Username: React.FC = () => {
+    return <Field name="username" type="text" component={FinalFormInput} fullWidth />;
+};
+
+const Name: React.FC = () => {
+    return <Field name="name" type="text" component={FinalFormInput} fullWidth />;
+};
+
+const RangeSliderField: React.FC = () => {
+    return (
+        <>
+            <Field name="range.minMaxArea" component={RangeSlider} fullWidth min={0} max={100} />
+            <Field name="range.minMaxHeight" component={RangeSlider} fullWidth min={0} max={100} />
+        </>
+    );
+};
+const HomePage: React.FC = () => {
+    return (
+        <>
+            <Field name="website.url" type="text" component={FinalFormInput} fullWidth />
+            <Field name="website.name" type="text" component={FinalFormInput} fullWidth />
+        </>
+    );
+};
+
+const ExampleWithSelect: React.FC = () => {
+    const options = [
+        { value: "Sincere@april.biz", label: "Sincere@april.biz" },
+        { value: "Shanna@melissa.tv", label: "Shanna@melissa.tv" },
+        { value: "Nathan@yesenia.net", label: "Nathan@yesenia.net" },
+    ];
+    return <Field name="email" type="text" component={FinalFormReactSelectStaticOptions} fullWidth options={options} />;
+};
+
+const fields: IFilterBarField[] = [
+    {
+        name: "username",
+        label: "Username",
+        component: Username,
+    },
+    {
+        name: "name",
+        label: "Name",
+        component: Name,
+    },
+    {
+        name: "range",
+        label: "RangeSlider",
+        component: RangeSliderField,
+    },
+    { name: "email", label: "Email Select", component: ExampleWithSelect },
+    {
+        name: "website",
+        label: "Homepage",
+        component: HomePage,
+    },
+];
+
+function Story() {
+    const filterApi = useTableQueryFilter<Partial<IFilterValues>>({
+        range: { minMaxArea: { min: 0, max: 100 }, minMaxHeight: { min: 0, max: 100 } },
+    });
+
+    const { tableData, api, loading, error } = useTableQuery<IQueryData, Partial<IVariables>>()(query, {
+        variables: {
+            ...filterApi.current,
+            pathFunction,
+        },
+        resolveTableData: (data) => ({
+            data: data.users,
+            totalCount: data.users.length,
+        }),
+    });
+
+    return (
+        <TableQuery api={api} loading={loading} error={error}>
+            <TableFilterFinalForm filterApi={filterApi}>
```

Comment:
What do you think of an API which uses JSX instead an `IFilterBarField[]`-object  to build the filterbar?
I think of something like this:
(They names I use in the example for the components are arbitrary.)

```
<TableFilterFinalForm filterApi={filterApi}>
         <Typography variant="h5">FilterBar</Typography>
          <FilterBar fieldBarWidth={150}>
                  <FilterBar.PopoverField label="Username">
                          <UserName/>
                  <FilterBar.PopoverField>
                  <FilterBar.PopoverField label="Name">
                          <Field name="name" type="text" component={FinalFormInput} fullWidth />
                  <FilterBar.PopoverField>
                <FilterBar.BehindMoreButton>
                      <FilterBar.PopoverField label="RangeSlider">
                               <RangeSliderField />
                               {
                                           user.isAdmin  &&  <AnotherRangeSliderFieldForAdmin />
                               }
                       <FilterBar.PopoverField>
                </FilterBar.BehindMoreButton   
          </FilterBar>
   </TableFilterFinalForm>

```

- Visualisation-logic like `maxCountInitialShown` can be well represented in JSX `<BehindMoreButton/>`.
- Better code readability
- Easier composition of components (see `user.isAdmin` - example)

Reply:
When i began with the first version of the component, we also thought about this approach. But i can't remember the reason why we actually have chosen the Object Array Approach. I think we have to discuss this in our group :)

Reply:
yes, much better api. I think you should experiment in that direction

Reply:
I also prefer the JSX-approach over the object approach 👍🏼